drivers/exec+java: Add configuration to restore previous PID/IPC namespace behavior (#9982)
hashicorp/nomad: website/content/docs/drivers/exec.mdx

## Plugin Options

- `default_pid_mode` `(string: optional)` - Defaults to `"private"`. Set to

>  `"private"` to enable namespace isolation for tasks by default, or `"host"` to

Reviewer: ```suggestion
 `"private"` to enable PID namespace isolation for tasks by default, or `"host"` to
```

 disable isolation.

!> **Warning:** If set to `"host"`, other processes running as the same user will
 be able to access sensitive process information like environment variables.

- `default_ipc_mode` `(string: optional)` - Defaults to `"private"`. Set to

> `"private"` to enable inter-process-communication isolation for tasks by default,

Reviewer: ```suggestion
`"private"` to enable IPC namespace isolation for tasks by default,
```

 or `"host"` to disable isolation.

!> **Warning:** If set to `"host"`, other processes running as the same user will be

>  able to make use of IPC features, like sending unexpected posix signals.

Reviewer: ```suggestion
 able to make use of IPC features, like sending unexpected POSIX signals.
```
